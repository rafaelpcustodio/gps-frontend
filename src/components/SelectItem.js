import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

import { borders } from '../helpers/border'
import { colors } from '../helpers/colors'

const StyledSelectItem = styled.li`
    border-radius: ${borders.default};
    color: ${colors.text};
    font-size: .75rem;
    list-style: none;
    padding: .25rem .5rem;
    white-space: nowrap;
    &:hover {
        background: ${colors.fourth};
        cursor: pointer;
    }
`

const SelectItem = ({ dataCy, onClick, option }) => (

    <StyledSelectItem onClick={ () => onClick(option) }>
        { option ? `${option}` : 'failure' }
    </StyledSelectItem>

)

SelectItem.propTypes = {
    onClick: PropTypes.func.isRequired,
    option: PropTypes.string.isRequired
}

export default SelectItem
