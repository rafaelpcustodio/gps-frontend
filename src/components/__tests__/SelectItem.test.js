import configureMockStore from 'redux-mock-store'
import { Map, Record } from 'immutable'
import React from 'react'
import renderer from 'react-test-renderer'

import { Provider } from 'react-redux'

import SelectItem from '../SelectItem'

describe('Tests for SelectItem component', () => {
    let store = {}
    let mockStore = {}

    it('Should render without crash', () => {
        mockStore = configureMockStore()
        store = mockStore(Record({
            gpsReducers: Map({
            })
        })())

        const tree = renderer.create(
            <Provider store={ store }>
                <SelectItem
                    option={''}
                    onClick={ jest.fn() }
                />
            </Provider>
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })
})
