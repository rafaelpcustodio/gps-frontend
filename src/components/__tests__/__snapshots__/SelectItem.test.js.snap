// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for SelectItem component Should render without crash 1`] = `
<li
  className="SelectItem__StyledSelectItem-c7nnhy-0 lkKUay"
  onClick={[Function]}
>
  error
</li>
`;
